proposals/ref-struct-interfaces.md

```diff
@@ -0,0 +1,195 @@
+
+```csharp
+// Proposed
+void M<T>(T p)
+    where T : IDisposable
+    allow T : ref struct
+{
+    p.Dispose();
+}
+
+// Alternative
+void M<T>(T p)
+    where T : IDisposable, ~ref struct
+{
+    p.Dispose();
+}
+```
+
+### Co and contra variance
+To be maximally useful type parameters that are `allow T : ref struct` must be compatible with generic variance. Specifically it must be legal for a parameter to be both co/contravariant and also `allow T: ref struct`. Lacking that they would not be usable in many of the most popular `delegate` and `interface` types in .NET like `Func<T>`, `Action<T>`, `IEnumerable<T>`, etc ...
+
+Given there is no actual variance when `struct` are involved these should be compatible. There is still some concern that I'm missing deeply generic variance cases. Need to sit down with @agocke to work out if this is truly safe or if there are deeply generic scenarios that need to be worked out.
+
+### Auto-applying to delegate members
+**Decision**: do not auto-apply
+
+For many generic `delegate` members the language could automatically apply `allow T: ref struct` as it's purely an upside change. Consider that for `Func<> / Action<>` style delegates there is no downside to expanding to allowing `ref struct`. The language can outline rules where it is safe to automatically apply this anti-constraint. This removes the manual process and would speed up the adoption of this feature.
+
+While that is true it can present a problem in multi-targeted scenarios. Code would compile in one target framework but fail in another. This could lead to confusion with customers and result in a desire for a more explicit opt-in.
+
+### Binary breaking change
+Adding `allow T: ref struct` to an existing API is not a source breaking change. It is purely expanding the set of allowed types for an API. Need to track down if this is a binary breaking change or not. Unclear if updating the attributes of a generic parameter constitute a binary breaking change.
+
+## Considerations
+
+### Runtime support
+This feature requires several pieces of support from the runtime / libraries team: 
+- Preventing default interface methods from applying to `ref struct`
+- API in `System.Reflection.Metadata` for encoding the `gpAcceptByRefLike` value.
+- Support for generic parameters being a `ref struct`
+
+Most of this support is likely already in place. The general `ref struct` as generic parameter support is already implemented as described [here](https://github.com/dotnet/runtime/blob/main/docs/design/features/byreflike-generics.md). It's possible the DIM implementation already account for `ref struct`. But each of these items needs to be tracked down.
+
+### Span<Span<T>>
```

Comment:
```suggestion
### Span&lt;Span&lt;T&gt;&gt;
```

Reply:
What does this change do? Today it renders as I would expect so not sure what this change brings.

Reply:
The title is currently rendered as `Span<Span>` where the nested `<T>` is missing.